pkg/component/kubernetes/controllermanager/controllermanager_test.go

```diff
@@ -1039,12 +1044,22 @@ func commandForKubernetesVersion(
 			command = append(command, fmt.Sprintf("--concurrent-statefulset-syncs=%d", *v))
 		}
 	} else {
-		controllers = append(controllers,
-			"-attachdetach",
-			"-cloud-node-lifecycle",
-			"-endpoint",
-			"-ephemeral-volume",
-		)
+		if versionutils.ConstraintK8sGreaterEqual133.Check(version) {
+			controllers = append(controllers,
+				"-attachdetach",
+				"-cloud-node-lifecycle",
+				"-device-taint-eviction-controller",
+				"-endpoint",
+				"-ephemeral-volume",
+			)
+		} else {
+			controllers = append(controllers,
+				"-attachdetach",
+				"-cloud-node-lifecycle",
+				"-endpoint",
+				"-ephemeral-volume",
+			)
+		}
```

Comment:
nit/opinionated: We could avoid a bit of repetition
```suggestion
		controllers = append(controllers,
			"-attachdetach",
			"-cloud-node-lifecycle",
			"-endpoint",
			"-ephemeral-volume",
		)
		if versionutils.ConstraintK8sGreaterEqual133.Check(version) {
			controllers = append(controllers,
				"-device-taint-eviction-controller",
			)
		}
```

Reply:
The problem with this approach would be that in the actual code the controllers are a set that eventually gets evaluated into an alphabetically ordered string and therefore the order of the `append` calls here is important - otherwise the test would fail, because of the different placement of the `-device-taint-eviction-controller`:
```
expected:
"--controllers=*,bootstrapsigner,tokencleaner,-attachdetach,-cloud-node-lifecycle,-device-taint-eviction-controller,-endpoint,-ephemeral-volume,-namespace,-nodeipam,-nodelifecycle,-persistentvolume-binder,-persistentvolume-expander,-pv-protection,-resourcequota,-serviceaccount-token,-ttl",

```
vs
```
got:
"--controllers=*,bootstrapsigner,tokencleaner,-attachdetach,-cloud-node-lifecycle,-endpoint,-ephemeral-volume,-device-taint-eviction-controller,-namespace,-nodeipam,-nodelifecycle,-persistentvolume-binder,-persistentvolume-expander,-pv-protection,-resourcequota,-serviceaccount-token,-ttl",
```

Reply:
I almost had a feeling this might be a case, thanks! Let's keep it like it is :)

Reply:
Still, my suggestion would be:
```diff
diff --git a/pkg/component/kubernetes/controllermanager/controllermanager_test.go b/pkg/component/kubernetes/controllermanager/controllermanager_test.go
index 4c762edf83..52296b3a62 100644
--- a/pkg/component/kubernetes/controllermanager/controllermanager_test.go
+++ b/pkg/component/kubernetes/controllermanager/controllermanager_test.go
@@ -1044,23 +1044,20 @@ func commandForKubernetesVersion(
 			command = append(command, fmt.Sprintf("--concurrent-statefulset-syncs=%d", *v))
 		}
 	} else {
+		controllers = append(controllers,
+			"-attachdetach",
+			"-cloud-node-lifecycle",
+		)
+
 		if versionutils.ConstraintK8sGreaterEqual133.Check(version) {
-			controllers = append(controllers,
-				"-attachdetach",
-				"-cloud-node-lifecycle",
-				"-device-taint-eviction-controller",
-				"-endpoint",
-				"-ephemeral-volume",
-			)
-		} else {
-			controllers = append(controllers,
-				"-attachdetach",
-				"-cloud-node-lifecycle",
-				"-endpoint",
-				"-ephemeral-volume",
-			)
+			controllers = append(controllers, "-device-taint-eviction-controller")
 		}
 
+		controllers = append(controllers,
+			"-endpoint",
+			"-ephemeral-volume",
+		)
+
 		if controllerWorkers.Namespace != nil && *controllerWorkers.Namespace == 0 {
 			controllers = append(controllers, "-namespace")
 		}

```